rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportService.java

```diff
@@ -437,4 +440,39 @@ public ByteBuffer getProjectReleaseSpreadSheetWithEcc(User user, String projectI
         }
         return ByteBuffer.wrap(IOUtils.toByteArray(exporter.makeExcelExport(releases)));
     }
+
+    public String getProjectSBOMBuffer(User user, String projectId, String bomType, boolean withSubProject) throws TException {
+        String bomString = "";
+            if (CommonUtils.isNotNullEmptyOrWhitespace(projectId)) {
+                RequestSummary summary = projectclient.exportCycloneDxSbom(projectId, bomType, withSubProject, user);
+                RequestStatus status = summary.getRequestStatus();
+                if (RequestStatus.FAILED_SANITY_CHECK.equals(status)) {
+                    bomString = status.name();
+                    throw new TException(bomString);
```

Comment:
Please throw application exception `SW360Exception` instead of Thrift's `TException`.